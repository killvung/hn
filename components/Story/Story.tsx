import React from 'react';
import { StoryProps } from '../../models/props';

const Story: React.FC<StoryProps> = ({ title, url, score, author, timestamp, descendants, rootId }) => {

    return (
        <div>
            Story Placeholder
            {title} {url} {score} {author} {timestamp} {descendants} {rootId}
        </div>
    );
}

export default Story;
