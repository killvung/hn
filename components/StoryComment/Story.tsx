import React from 'react';

const Story: React.FC = () => (
    <div>Story Placeholder</div>
);

export default Story;
