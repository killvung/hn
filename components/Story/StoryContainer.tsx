import React, { useState, useEffect } from 'react';
import { useRouter } from 'next/router'
import Story from './Story';
import { storyDataToStoryProps } from '../../models/mapper'

import fetchPost from '../../utils/fetchPost';

const StoryContainer: React.FC = () => {
    // TODO: Change type
    const [story, setStory] = useState<any>();
    const router = useRouter();

    useEffect(() => {
        const storyId = router.query.storyId as string;
        if (storyId) {
            fetchPost(storyId).then((data: any) => {
                setStory(storyDataToStoryProps(data));
            })
        }
    }, [router.query.storyId]);

    return (
        <Story {...story} />
    );
}

export default StoryContainer;
