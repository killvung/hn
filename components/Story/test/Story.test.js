import React from 'react';
import Story from '../Story';

test('Story - Title, points, author, date, comment countexisted', () => {

});
