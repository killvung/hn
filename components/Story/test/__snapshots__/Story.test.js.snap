// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story - Title, points, author, date, comment count existed 1`] = `
<div>
  Story Placeholder
   
   
   
   
   
   
</div>
`;
